packages/puppeteer-extra-plugin-stealth/evasions/navigator.hardwareConcurrency/index.js

```diff
@@ -0,0 +1,32 @@
+'use strict'
+
+const {PuppeteerExtraPlugin} = require('puppeteer-extra-plugin')
+
+/**
+ * Allows faking CPU core number. This allows creating different profiles.
+ * Will overwrite `navigator.hardwareConcurrency` property.
+ */
+class Plugin extends PuppeteerExtraPlugin {
+    constructor(opts = {}) {
+        super(opts)
+    }
+
+    get name() {
+        return 'stealth/evasions/navigator.hardwareConcurrency'
+    }
+
+
+    async onPageCreated(page) {
+    let cores = this.opts.cores || false;
+        await page.evaluateOnNewDocument((cores) => {
+            let originalCores = navigator.hardwareConcurrency;
+             cores = cores || originalCores;
+            delete Object.getPrototypeOf(navigator).hardwareConcurrency;
```

Comment:
Is there a specific reason to delete and then set? I think only setting should be sufficient.

Reply:
yeah it seemed hardwareConcurreny is readonly. I could not reset its value.
Its important to note, this changes the order of properties of navigator. Something that was being flagged in an amIunique website

Reply:
Can you try this instead?
```js
Object.defineProperty(Object.getPrototypeOf(navigator), 'hardwareConcurrency', { get: () => cores || originalCores });
```
That should do the job

Reply:
Can you try to overwrite is @gyaaniguy? That should resolve the order of properties issue.

Reply:
yeah sure. Sorry I got pre-occupied. This project didn't come to fruition.